api/src/main/java/ca/bc/gov/educ/api/grad/report/config/RestWebClient.java

```diff
@@ -2,78 +2,89 @@
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
 import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
 import org.springframework.web.reactive.function.client.ExchangeStrategies;
 import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.DefaultUriBuilderFactory;
+import reactor.netty.http.client.HttpClient;
 
 @Configuration
 @Profile("!test")
 public class RestWebClient {
+    private final DefaultUriBuilderFactory factory;
+    private final ClientHttpConnector connector;
     EducGradReportApiConstants constants;
 
     @Autowired
     public RestWebClient(EducGradReportApiConstants constants) {
         this.constants = constants;
+        this.factory = new DefaultUriBuilderFactory();
+        this.factory.setEncodingMode(DefaultUriBuilderFactory.EncodingMode.NONE);
+        final HttpClient client = HttpClient.create().compress(true);
+        client.warmup()
+            .block();
+        this.connector = new ReactorClientHttpConnector(client);
     }
+
     @Bean("graduationReportClient")
-    public WebClient getGraduationReportClient(OAuth2AuthorizedClientManager authorizedClientManager) {
-        ServletOAuth2AuthorizedClientExchangeFilterFunction filter = new ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager);
-        filter.setDefaultClientRegistrationId("graduation-report-client");
-        return WebClient.builder()
-                .exchangeStrategies(ExchangeStrategies
-                        .builder()
-                        .codecs(codecs -> codecs
-                                .defaultCodecs()
-                                .maxInMemorySize(50 * 1024 * 1024))
-                        .build())
-                .apply(filter.oauth2Configuration())
-                .filter(this.log())
-                .build();
-    }
-    @Bean
-    public OAuth2AuthorizedClientManager authorizedClientManager(
-            ClientRegistrationRepository clientRegistrationRepository,
-            OAuth2AuthorizedClientRepository authorizedClientRepository) {
-        OAuth2AuthorizedClientProvider authorizedClientProvider = OAuth2AuthorizedClientProviderBuilder.builder()
-                .clientCredentials()
-                .build();
-        DefaultOAuth2AuthorizedClientManager authorizedClientManager = new DefaultOAuth2AuthorizedClientManager(
-                clientRegistrationRepository, authorizedClientRepository);
-        authorizedClientManager.setAuthorizedClientProvider(authorizedClientProvider);
-        return authorizedClientManager;
+    @Autowired
+    WebClient getGraduationReportClient(final WebClient.Builder builder) {
+        val clientRegistryRepo = new InMemoryReactiveClientRegistrationRepository(ClientRegistration
+            .withRegistrationId(this.constants.getUserName())
```

Comment:
For the graduationReportClient, there are different credentials which are set under the security.oauth2.client property in the application.yaml file. This is the preferred client moving forward. The client was set previously with filter.setDefaultClientRegistrationId("graduation-report-client"); If you are unable to do this with the new setup here, then please retrieve the client-id and key from the values under security.oauth.client in application.yaml.